### ISSUE TITLE
cmd.run state with pgrep unless statement outputs differently on salt-master and local

### ISSUE THREAD

**Issue by wesley-c42** [saltstack/salt#56516]
### Description of Issue
When running a cmd.run state with a pgrep unless statement, I get a different outputs locally and on the salt-master.

### Setup
```
Install the Cloud Monitoring agent:
  cmd.run:
    - name: >
        curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ;
        bash install-monitoring-agent.sh
    - unless: pgrep -f collectd
```

### Steps to Reproduce Issue
State output when ran on salt-master:
```
----------
          ID: Install the Cloud Monitoring agent
    Function: cmd.run
        Name: curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ; bash install-monitoring-agent.sh

      Result: True
     Comment: unless condition is true
     Started: 10:17:54.422964
    Duration: 13.184 ms
     Changes:
----------
```

State output when ran locally:
```
----------
          ID: Install the Cloud Monitoring agent
    Function: cmd.run
        Name: curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ; bash install-monitoring-agent.sh

      Result: None
     Comment: Command "curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ; bash install-monitoring-agent.sh
              " would have been executed
     Started: 10:18:12.852482
    Duration: 10.704 ms
     Changes:
----------
```

### Versions Report

```
Salt Version:
           Salt: 2018.3.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.8
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.7
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 2.7.17 (default, Nov  7 2019, 10:07:09)
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-91-generic
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by DmitryKuzmenko** [comment]
@wesley-c42 what do you mean saying: "ran localy"? Could you please describe your environment and what commands you use on each case?
I've tried to ran your state with `salt '*' state.apply <state_id>`, `salt-call state.apply <state_id>` and `salt-call --local state.apply <state_id>`. The last one tried with and without salt daemons running. The same result on all:
```
          ID: Install the Cloud Monitoring agent
    Function: cmd.run
        Name: curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ; bash install-monitoring-agent.sh

      Result: True
     Comment: unless condition is true
     Started: 14:48:05.237128
    Duration: 76.931 ms
     Changes:
```

**Comment by sagetherage** [comment]
@wesley-c42 can you give more details, please?

**Comment by wesley-c42** [comment]
@sagetherage @DmitryKuzmenko Sorry for the late reply.
By locally I meant `salt-call state.apply <state_id>`. Both salt-master and minion are ubuntu 18.04.
I had worked around it by using grep, which didn't give the issue:
```
Install the Cloud Monitoring agent:
  cmd.run:
    - name: >
        curl -sSO https://dl.google.com/cloudagents/install-monitoring-agent.sh ;
        bash install-monitoring-agent.sh
    - unless: ps -ef|grep collectd|grep -v grep
```

Since then I've updated to 3000.3, tested this again just now and I no longer see this issue with pgrep.

**Comment by sagetherage** [comment]
@wesley-c42 great news that it is fixed in a newer version! I am closing the ticket as won't fix for now in 2018.3.4 and corrected by v3000.3, please let us know if you see this again, please. Thank you!
